### PR TITLE
created repo for dol

### PR DESCRIPTION
**Pull request by marneneha**
did chenges in CMakelist pkg.xml launch file

### CODE REVIEW

**Comment by marneneha** [comment]
tested by giving multiple runtime goals

File: launch/maxl_DOL.launch [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+    <arg name="mobileGoalTopic" default="/move_base_simple/goal"/> 
+    <arg name="loop_rate_value" default="10"/>
+    <arg name="goal_dist_tolerance" default = "0.1"/> <!-- in meters -->
+    <arg name="rel_motion_mode" default = "false"/>
+    <arg name="map_frame" default="map"/>
+    <arg name="robot_frame" default="base_link"/>
+    <arg name="dol_params" default="$(find mw_maxl_planner)/config/dol_config"/>
+
+    <!-- =============== DOL =============== -->
+    <node pkg="mw_maxl_planner" name="mw_maxl_dol_node" type="mw_maxl_dol_node" output="screen">
```
Comment (puru07) [inline launch/maxl_DOL.launch]: this will not work. 
this should be launching robot follower node

File: launch/maxl_DOL.launch [hunk 2]
```diff
@@ -0,0 +1,24 @@
+<?xml version="1.0"?>
```
Comment (puru07) [inline launch/maxl_DOL.launch]: rename this file to robot_follower.launch

File: src/robot_follower_node.cpp [hunk 3]
```diff
@@ -0,0 +1,200 @@
+#include <mw_maxl_planner/mw_maxl_planner_ros1.h>
+
+class DOL
```
Comment (puru07) [inline src/robot_follower_node.cpp]: rename the class, remove DOL, put robot_follower

File: src/robot_follower_node.cpp [hunk 4]
```diff
@@ -0,0 +1,200 @@
...
+        if (mw_maxl->isGoalReached(goalXYTolerance_, dummy_angle_goal_tolerance))
+        {
+            return true;
+        }
+        return false;
+    }
+};
+
+int main(int argc, char **argv) {
+    ros::init(argc, argv, "maxl_dol_node");
```
Comment (puru07) [inline src/robot_follower_node.cpp]: robot_follower_node

**Review by puru07 — COMMENTED** [review]
some changes in the name

File: config/dol_config/mw_maxl_planner.yaml [hunk 5]
```diff
@@ -0,0 +1,81 @@
+mw_maxl_planner:
```
Comment (puru07) [inline config/dol_config/mw_maxl_planner.yaml]: change the name of the folder, it should be robot_follower_config ( folder name)

File: package.xml [hunk 6]
```diff
@@ -0,0 +1,65 @@
+<?xml version="1.0"?>
+<package format="2">
+  <name>robot_follower</name>
+  <version>0.0.0</version>
+  <description>The robot_follower package</description>
+
+  <!-- One maintainer tag required, multiple allowed, one person per tag -->
+  <!-- Example:  -->
+  <!-- <maintainer email="jane.doe@example.com">Jane Doe</maintainer> -->
+  <maintainer email="neha@todo.todo">neha</maintainer>
```
Comment (puru07) [inline package.xml]: your correct email

File: package.xml [hunk 7]
```diff
@@ -0,0 +1,65 @@
...
+
+
+  <!-- One license tag required, multiple allowed, one license per tag -->
+  <!-- Commonly used license strings: -->
+  <!--   BSD, MIT, Boost Software License, GPLv2, GPLv3, LGPLv2.1, LGPLv3 -->
+  <license>TODO</license>
```
Comment (puru07) [inline package.xml]: BSD

File: src/robot_follower_node.cpp [hunk 8]
```diff
@@ -0,0 +1,200 @@
+#include <mw_maxl_planner/mw_maxl_planner_ros1.h>
```
Comment (puru07) [inline src/robot_follower_node.cpp:9]: put meta info, 
for example:
file name:
brief: ( description of the purpose of the file)
author: ( name and email)
date: 
company:

**Pull request merged**